### PR TITLE
feat(data-service): Improves identification of atlas cluster

### PR DESCRIPTION
**Pull request by leorossi**
### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/instance-model/lib/model.js [hunk 1]
```diff
@@ -105,14 +105,9 @@ const InstanceModel = AmpersandModel.extend(
       databasesStatusError: { type: 'string', default: null },
       refreshingStatus: { type: 'string', default: 'initial' },
       refreshingStatusError: { type: 'string', default: null },
+      isAtla: { tyep: 'boolean', default: false }
```
Comment (gribnoysup) [inline packages/instance-model/lib/model.js:108]:
```suggestion
      isAtlas: { tyep: 'boolean', default: false }
```

File: packages/data-service/src/instance-detail-helper.spec.ts [hunk 2]
```diff
@@ -89,6 +89,10 @@ describe('instance-detail-helper', function () {
             null
           );
         });
+
+        it('should not be identified as atlas', function () {
+          expect(instanceDetails).to.have.property('isAtlas', false);
+        });
```
Comment (gribnoysup) [inline packages/data-service/src/instance-detail-helper.spec.ts:95]: Maybe it makes sense to add a few test cases with the mocked client below to check the outcome where command succeeds?

File: packages/data-service/src/instance-detail-helper.ts [hunk 3]
```diff
@@ -88,6 +89,7 @@ export type InstanceDetails = {
   genuineMongoDB: GenuineMongoDBDetails;
   dataLake: DataLakeDetails;
   featureCompatibilityVersion: string | null;
+  isAtlas: boolean;
```
Comment (mcasimir) [inline packages/data-service/src/instance-detail-helper.ts:92]: should this be used here? https://github.com/mongodb-js/compass/pull/2589/files#diff-0a7ade426ce694c5c7bd013c0f5f85f445376c0210d589cc86c704bf9db5e3f9R27

only after the connection happens?

File: packages/data-service/src/instance-detail-helper.ts [hunk 4]
```diff
@@ -134,9 +141,22 @@ export async function getInstance(
...
   };
 }
 
+function checkIsAtlas(
+  client: MongoClient,
+  atlasVersionInfo: AtlasVersionInfo
+): boolean {
+  const firstHost = client.options.hosts[0]?.host || '';
+
+  if (undefined === atlasVersionInfo.version) {
```
Comment (mcasimir) [inline packages/data-service/src/instance-detail-helper.ts:154]: non really important but seems that in the rest of the codebase we do `something === undefined`, if you want to keep it consistent

**Comment by leorossi** [comment]
Sorry guys I meant to open a draft PR because of many open points 🙏

**Comment by gribnoysup** [comment]
Just a reminder not to forget to use this new `isAtlas` value for the telemetry so we can also address COMPASS-5323 with this PR

**Comment by mcasimir** [comment]
> Just a reminder not to forget to use this new `isAtlas` value for the telemetry so we can also address [COMPASS-5323](https://jira.mongodb.org/projects/COMPASS/issues/COMPASS-5323) with this PR

We probably want to wait for this to be merged first: https://github.com/mongodb-js/compass/pull/2589

File: packages/data-service/src/instance-detail-helper.ts [hunk 5]
```diff
@@ -134,9 +141,23 @@ export async function getInstance(
...
       getParameterResult?.featureCompatibilityVersion.version ?? null,
+    isAtlas: checkIsAtlas(client, atlasVersionResult),
   };
 }
 
+function checkIsAtlas(
+  client: MongoClient,
+  atlasVersionInfo: AtlasVersionInfo
+): boolean {
+  console.log(client.options.hosts);
```
Comment (gribnoysup) [inline packages/data-service/src/instance-detail-helper.ts:152]: This is probably an accidental leftover, right?
```suggestion
```

**Review by gribnoysup — COMMENTED** [review]
Looks good to me! Even though old analytics is going away soon, we probably want to also fix it and use the new `isAtlas` value there too:

https://github.com/mongodb-js/compass/blob/200fd6a31e0245b38c300e3c9add30c5b4070ef5/packages/compass-metrics/src/modules/rules.js#L86

File: packages/data-service/src/instance-detail-helper.ts [hunk 6]
```diff
@@ -134,9 +141,23 @@ export async function getInstance(
...
+  const firstHost = client.options.hosts[0]?.host || '';
+
+  if (atlasVersionInfo.version === '') {
+    return /mongodb(-dev)?.net$/i.test(firstHost);
```
Comment (addaleax) [inline packages/data-service/src/instance-detail-helper.ts:156]:
```suggestion
    return /mongodb(-dev)?\.net$/i.test(firstHost);
```

**Review by addaleax — APPROVED** [review]

File: packages/data-service/src/instance-detail-helper.spec.ts [hunk 7]
```diff
@@ -241,6 +245,53 @@ describe('instance-detail-helper', function () {
...
         );
       });
+      it(`should be identified as atlas with hostname correct hostnames`, function () {
+        ['myserver.mongodb.net', 'myserver.mongodb-dev.net'].map(
+          async (hostname) => {
+            const client = createMongoClientMock({
+              hosts: [{ host: hostname, port: 9999 }],
+              commands: {
+                buildInfo: {},
+                getCmdLineOpts: fixtures.DOCUMENTDB_CMD_LINE_OPTS,
```
Comment (mcasimir) [inline packages/data-service/src/instance-detail-helper.spec.ts:255]: should not affect the test, but we could use `CMD_LINE_OPTS` to avoid confusion.

**Review by mcasimir — APPROVED** [review]
nice!

**Pull request merged**